yb-voyager/src/query/queryissue/parser_issue_detector.go

```diff
@@ -467,6 +510,33 @@ func (p *ParserIssueDetector) FinalizeColumnMetadata() {
 
 	p.populateColumnMetadataDerivedVars()
 
+	// Build partition hierarchies
+	p.buildPartitionHierarchies()
+}
+
+// buildPartitionHierarchies builds the direct child relationships for all tables
+func (p *ParserIssueDetector) buildPartitionHierarchies() {
+	// Initialize Partitions slice for all tables
+	for _, tm := range p.tablesMetadata {
+		tm.Partitions = []*TableMetadata{}
+	}
+
+	// Build parent-child relationships
+	for _, tm := range p.tablesMetadata {
+		if tm.PartitionedFrom != "" {
+			// Find parent table
+			for _, parentTM := range p.tablesMetadata {
+				// Compare both qualified and unqualified names since PartitionedFrom can be either.
+				// CREATE TABLE ... PARTITION OF produces unqualified names (e.g., "sales") because
+				// PostgreSQL doesn't include schema info in parse tree, while ALTER TABLE ... ATTACH PARTITION
+				// produces qualified names (e.g., "public.sales") when schema is explicitly specified.
+				if parentTM.TableName == tm.PartitionedFrom || parentTM.GetObjectName() == tm.PartitionedFrom {
```

Comment:
Add TODO to use `sqlname` for proper naming handling for qualified / unqualified names